src/redux/books/booksSlice.js

```diff
@@ -1,22 +1,23 @@
 import { createSlice } from '@reduxjs/toolkit';
 
-const initialState = [];
+const initialState = {
+  book: [],
+};
 
 const bookSlice = createSlice({
-  name: 'book',
+  name: 'books',
   initialState,
   reducers: {
-    add: (state, action) => {
-      state.push(action.payload);
+    addbook: (state, action) => {
+      state.book.push(action.payload);
     },
-    remove: (state, action) => {
-      const newstate = state.filter((item) => item.id !== action.payload);
-      const updateIndex = newstate.map((item, index) => ({ ...item, id: index + 1 }));
-      return updateIndex;
+    removebook: (state, action) => {
+      const newstate = state.book.filter((item) => item.id !== action.payload);
+      return { book: newstate };
```

Comment:
- [ ] Please note that when you use the `createSlice` function provided by the Redux Toolkit, the reducers are responsible for updating the state based on the current action. Therefore, you don't need to return the new state from the reducer as suggested by the previous reviewer. Instead, you can update the state directly within the reducer function, as shown in the example below: 👇
```jsx
removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```

Reply:
How does is affect because returning anything with modify the state completely.
as it not mutating array right now but if i use splice method then i have to return so how does it affect whether is save it in state.books or return whole the filtered array as state.
Can you elaborate on this?

Reply:
- Hello @vikas924, your question is a valid one. The reason I suggested making that change is that in your next milestone, you'll be dispatching actions that call the `removeBook` reducer with the ID of the book to be removed. In that case, you don't need to return any value from the reducer, but instead, you'll want it to update the state directly. By removing the return statement and updating the state directly in the reducer, you'll be able to simplify your code and make it more efficient. Let me know if you have any further questions!

Reply:
- And as per this comment "as it not mutating array right now but if i use splice method then i have to return " please revisit your code as you're using the filter method and not splice and as you know the filter method returns a new array and doesn't mutate the original array so that means you'll have to assign the returned array from the filter method to the initial state so it gets updated.

Reply:
yeah i get it will be more efficient to save it in state rather than returning whole state everytime but it could be optional suggestion as it is not requirment for this project. 
As for mutating array i just asked to know which is more efficient approach
Thanks you for your feedback

Reply:
@vikas924 One of the project requirements is to implement [Redux Toolkit](https://redux-toolkit.js.org/) inside of a React application. And as per the guideline [found here](https://redux-toolkit.js.org/api/createSlice#reducers) I'd advise you to use similar patterns in the future but please note the distinction between the normal redux way and the redux-toolkit way.